### PR TITLE
Bs 59

### LINKED ISSUE #59: Создание Onboarding UI

### PR DESCRIPTION
**Pull request by antonhoang**
## Details 
- implement  onboardingUI

## Video
https://user-images.githubusercontent.com/62542442/184037001-0e6cb149-e9ee-4cc0-9b6c-09d5e34f9493.mov

### CODE REVIEW

**Comment by VladimirBrejcha** [comment]
I'm not 100% happy with onboarding right now. It looks great, but when I swipe, I always see this line between images which is not very beautiful I think..

image

I do have idea however. Maybe try to experiment with cross-fade animation on Onboarding instead of swipes? It could look pretty cool IMO

File: bonsai/Modules/Onboarding/Onboarding.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        "bonsai_1",
+        "bonsai_2",
+        "bonsai_3"
+    ]
+    
+    var body: some View {
+        ZStack {
+            VStack {
+                TabView(selection: $selection) {
+                    ForEach(0..<3) { index in
```
Comment (VladimirBrejcha) [inline bonsai/Modules/Onboarding/Onboarding.swift]: why not use `0..<onboardingImages.count` instead of 0..<3

Comment (VladimirBrejcha) [reply bonsai/Modules/Onboarding/Onboarding.swift]: or maybe ForEach(onboardingImages) could be even better? Not sure if it will work tho

Comment (antonhoang) [reply bonsai/Modules/Onboarding/Onboarding.swift]: updated+

**Comment by antonhoang** [comment]
> I'm not 100% happy with onboarding right now. It looks great, but when I swipe, I always see this line between images which is not very beautiful I think..
> 
> image
> 
> I do have idea however. Maybe try to experiment with cross-fade animation on Onboarding instead of swipes? It could look pretty cool IMO

try to do but seems cringe, i think need to discuss animation :)

File: bonsai/Modules/Onboarding/Onboarding.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            image: "bonsai_3",
+            title: "Where can I get some?",
+            description: "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable."),
+    ]
+    
+    var body: some View {
+        VStack {
+            ZStack {
+                
+                if selection == 0 {
```
Comment (Valter4578) [inline bonsai/Modules/Onboarding/Onboarding.swift:39]: Why did u do in such why? Isn't it easier to do just 
`OnboardingCard(dataSource: dataSource[selection])` and check if selection isn't greater that 2

**Review by Valter4578 — CHANGES_REQUESTED** [review]
In general really great work 👏

File: bonsai/Modules/Onboarding/Onboarding.swift (hunk 2 above)
Comment (VladimirBrejcha) [reply bonsai/Modules/Onboarding/Onboarding.swift:39]: Because of SwiftUI bugs. Trust me, we tried x) But this workaround seems needed

**Pull request merged**